cli/pcluster/commands.py

```diff
@@ -173,16 +219,27 @@ def create(args):  # noqa: C901 FIXME!!!
     _check_for_updates(pcluster_config)
 
     bucket_name = None
+    artifact_directory = None
     try:
         cfn_client = boto3.client("cloudformation")
         stack_name = utils.get_stack_name(args.cluster_name)
 
         # merge tags from configuration, command-line and internal ones
         tags = _evaluate_tags(pcluster_config, preferred_tags=args.tags)
 
-        bucket_name = _create_bucket_with_resources(pcluster_config, storage_data, tags)
-        if bucket_name:
-            cfn_params["ResourcesS3Bucket"] = bucket_name
+        bucket_name, artifact_directory, cleanup_bucket = _setup_bucket_with_resources(
```

Comment:
what error are we going to propagate if this function raises exception?

Reply:
`_setup_bucket_with_resources ` will log the specific error message already, and `create` will just log the exception and `exit(1)`. I think this should be ok?